internal/presenter/packages/spdx_json_presenter.go

```diff
@@ -39,7 +39,8 @@ func (pres *SPDXJsonPresenter) Present(output io.Writer) error {
 	return enc.Encode(&doc)
 }
 
-func newSPDXJsonDocument(catalog *pkg.Catalog, srcMetadata source.Metadata) spdx22.Document {
+// NewSPDXJsonDocument creates and populates a new JSON document struct that follows the SPDX 2.2 spec from the given cataloging results.
+func NewSPDXJsonDocument(catalog *pkg.Catalog, srcMetadata source.Metadata) spdx22.Document {
```

Comment:
Just curious: Why did this get exported?

Reply:
We export NewJsonDocument as it is consumed by the poweruser package. It seemed incorrect to have one Document constructor exposed while the other was private. I also think it's something that should have an official Documentation string highlighted above the function since its counterpart would also end up in the godoc if generated.

Reply:
Gotcha, interesting! My thoughts, in no particular order:

- 100% agree re: doc comments 🌟 
- Architectural question: Do we plan on creating SPDX documents for any non-package commands?
- My pref in general is not to export something unless it's being used internally in another package, or we expect it to be used by a consumer of our module. In this case, it looks like this code is under `internal/` so I don't think it would be consumed by another module. Is that right?

Reply:
lowercased - if we start doing anything in `poweruser` with the function we can export it then.